generator/generator.go

```diff
@@ -0,0 +1,163 @@
+package generator
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/helm/helm/log"
+)
+
+const GeneratorKeyword = "helm:generate "
+
+// Walk walks a chart directory and executes generators as it finds them.
+//
+// Returns the number of generators executed.
```

Comment:
can you add some docs on when errors will be returned, and what some of the common ones are?